util/proposal.go

```diff
@@ -34,5 +41,22 @@ func GetProposalName(payload string) (string, error) {
 
 // IsValidProposalName reports whether str is a valid proposal name
 func IsValidProposalName(str string) bool {
+	validProposalName := regexp.MustCompile(proposalNameRegExp)
```

Comment:
This should be moved to the global initialisation since we just need to do it once.